stdlib/public/runtime/ImageInspectionCommon.h

```diff
@@ -71,6 +71,20 @@ struct SectionInfo {
 SWIFT_RUNTIME_EXPORT
 void swift_addNewDSOImage(struct swift::MetadataSections *sections);
 
+/// Get all metadata sections in the current process that are known to the
+/// Swift runtime.
+///
+/// \param outCount On return, the number of elements in the resulting array.
+///
+/// \returns A C array of structures describing all known metadata sections.
+///   The caller should call \c free() to free this array when done.
+///
+/// On Mach-O-based platforms (i.e. Apple platforms), this function is
+/// unavailable. On those plaforms, use dyld API to enumerate loaded images and
+/// their corresponding metadata sections.
+SWIFT_RUNTIME_EXPORT
+swift::MetadataSections *swift_copyAllMetadataSections(size_t *outCount);
```

Comment:
If we're going to expose this call, maybe we should also provide an implementation that uses dyld for us. Not quite sure if that's the appropriate place for the dyld-using implementation though.

Reply:
Right now, `MetadataSections` is entirely unused on Apple platforms, so we'd need to synthesize the structures. We _can_ do that but in the absence of a need (and the presence of an alternate solution) I am hesitant.